### DISCUSSION TITLE
No Flows are Shown only Exporters and "Last flow" [troubleshooting commands outputs inside]

### DISCUSSION THREAD

**Discussion by ntrifonova** [akvorado/akvorado#1451] [Q&A] [answered]
![Screenshot 2024-10-16 at 10 05 34 AM](https://github.com/user-attachments/assets/32ee6c4c-9fac-4327-8d86-431301f8e191)

Outputs from troubleshooting commands:
```
# curl -s http://akvorado:8080/api/v0/inlet/flows\?limit\=1
{
    "TimeReceived": 1729066030,
    "SamplingRate": 8192,
    [...]
```

```
# curl -s http://akvorado:8080/api/v0/inlet/metrics | grep '^akvorado_inlet_flow_input_udp_packets'
akvorado_inlet_flow_input_udp_packets_total{exporter="IP1",listener="0.0.0.0:2055",worker="0"} 36999
akvorado_inlet_flow_input_udp_packets_total{exporter="IP2",listener="0.0.0.0:2055",worker="2"} 2683
akvorado_inlet_flow_input_udp_packets_total{exporter="IP3",listener="0.0.0.0:6343",worker="1"} 26945
akvorado_inlet_flow_input_udp_packets_total{exporter="IP4",listener="0.0.0.0:2055",worker="0"} 3308
akvorado_inlet_flow_input_udp_packets_total{exporter="IP5",listener="0.0.0.0:6343",worker="0"} 377475
akvorado_inlet_flow_input_udp_packets_total{exporter="IP6",listener="0.0.0.0:6343",worker="2"} 168407
```

```
# curl -s http://akvorado:8080/api/v0/inlet/metrics | grep '^akvorado_inlet_core_forwarded_flows_total'
akvorado_inlet_core_forwarded_flows_total{exporter="IP1"} 864986
akvorado_inlet_core_forwarded_flows_total{exporter="IP2"} 18762
akvorado_inlet_core_forwarded_flows_total{exporter="IP3"} 77158
akvorado_inlet_core_forwarded_flows_total{exporter="IP4"} 2.317698e+06
akvorado_inlet_core_forwarded_flows_total{exporter="IP5"} 1.000907e+06
akvorado_inlet_core_forwarded_flows_total{exporter="IP6"} 55229
```

```
# curl -s http://akvorado:8080/api/v0/inlet/metrics | grep '^akvorado_inlet_kafka_sent_messages_total'
akvorado_inlet_kafka_sent_messages_total{exporter="IP1"} 872235
akvorado_inlet_kafka_sent_messages_total{exporter="IP2"} 18857
akvorado_inlet_kafka_sent_messages_total{exporter="IP3"} 77890
akvorado_inlet_kafka_sent_messages_total{exporter="IP4"} 2.337609e+06
akvorado_inlet_kafka_sent_messages_total{exporter="IP5"} 1.008985e+06
akvorado_inlet_kafka_sent_messages_total{exporter="IP6"} 55616
```

```
# curl -s http://akvorado:8080/api/v0/inlet/metrics | grep '^akvorado_inlet' | grep _error
[some errors here but the count is not increasing]
```

Classifiers:
```
    exporter-classifiers:
      - ClassifySiteRegex(Exporter.Name, "^[^-]+-([^-]+)-", "$1")
      - ClassifyRegionRegex(Exporter.Name, "^([^-])+-[^-]+-", "$1")
      - ClassifyRole("ar")
    interface-classifiers:
      - |
        ClassifyConnectivityRegex(Interface.Description, "^(?i)(transit|pni|ppni|ix):? ", "$1") &&
        ClassifyProviderRegex(Interface.Description, "^\\S+?\\s(?:to )?(\\S+)", "$1") &&
        ClassifyExternal()
      - ClassifyInternal()
```
`InIfBoundary` is set to **Internal** for all exporters.

The last errors in Clickhouse:
```
Row 5:
──────
last_error_time:    2024-10-16 06:33:26
last_error_message: All commit attempts failed. Last block was already written to target table(s), but was not committed to Kafka.

Row 6:
──────
last_error_time:    2024-10-16 08:28:23
last_error_message: Invalid credentials

Row 7:
──────
last_error_time:    2024-10-16 08:28:23
last_error_message: default: Authentication failed: password is incorrect, or there is no user with such name.
```
(not sure why it says authentication failed, I logged in from the first attempt).

No errors in the logs of the services. What am I missing? Would appreciate help a lot! Thank you.

**Comment by ntrifonova** [comment]
We would really appreciate any help and the nudge to the right direction. We will be happy to use Akvorado and it will benefit us a lot...

**Reply by vincentbernat** [reply]
You had some flows in the past since you have exporters, but not anymore. Flows/s is the last 5 minutes.

Are the flows sent to Kafka increasing? Is the last flow on the home page recent? The last flow on the homepage is retrieved from ClickHouse. Often, in cases like yours, there is no more disk space.

**Reply by ntrifonova** [reply]
Thank you Vincent! Disk size is not an issue (yet). It is 55% used. Though it is growing very fast. 

> Are the flows sent to Kafka increasing?

Yes. I am looking at `akvorado_inlet_kafka_sent_messages_total`.

> Is the last flow on the home page recent?

No. I think here is the bottleneck somewhere. I noticed that kafka lag is fluctuating. Probably we have so many flows, that it is either kafka not processing flows fast enough, or Clickhouse is slow. Not sure. But we always have a lag (I am checking with `./kafka-consumer-groups.sh --bootstrap-server localhost:9092 --describe --group clickhouse`).

I increased Kafka workers first to 10. It was just an obvious fix(?) I think. Didn't help much. Now I increased it to 15. Maybe there is a way to increase number of consumers? Or somehow increase the speed of the processing pipeline?

Thank you for your help! We really would like to make it work :)

**Reply by vincentbernat** [reply, marked as answer]
Is the last flow as received by ClickHouse just "late" or it's stuck? If you don't loose packets in the inlet (check [documentation](https://demo.akvorado.net/docs/troubleshooting#dropped-packets-under-load)), it is unlikely this is the one that you need to tune. More likely, ClickHouse could be an issue. In configuration, try:

```yaml
clickhouse:
  kafka:
    consumers: 16
    engine-settings:
      - kafka_flush_interval_ms = 1000
kafka:
  topic-configuration:
    num-partitions: 16
    replication-factor: 1
```

If the lag is still present, you should look at ClickHouse logs. I have already seen it may pause to resynchronize.

**Reply by ntrifonova** [reply]
It definitely looks better now, and we have flows and a very small lag for each of the consumers!
